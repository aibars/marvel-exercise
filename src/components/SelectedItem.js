import React from 'react';
import '../styles/SelectedItem.css';
import '../styles/App.css';

export default function SelectedItem(props) {
    const { item, isFetching } = props.selectedItem;

    return (
        <div id="right-col" style={{ opacity: isFetching ? 0.5 : 1 }}>
            <label className="list-title"> Selected Pokemon:</label>
            {isFetching && <h2>Loading...</h2>}
            {item &&
                (<div className="beast-info">
                    <br />
                    <span className="selected-name">Beast Name: {item.species.name[0].toUpperCase() + item.species.name.slice(1)}</span>
                    <img width="150" height="150" src={item.sprites.front_default} alt={item.species.name} />
                    <br />
                    Abilities:
                    <ul className="abilities-list">
                        {item.abilities.map((elem, i) => (
                            <li key={i} className="ability">{elem.ability.name}</li>
                        ))}
                    </ul>
                    Stats:
                    <ul className="abilities-list">
                        {item.stats.map((elem, i) => (
                            <li key={i} className="ability">{elem.stat.name}</li>
                        ))}
                    </ul>
                </div>)}
        </div>
    );
}