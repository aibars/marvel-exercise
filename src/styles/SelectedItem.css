.selected-name {
    padding-top: 15px;
    text-align: center;
    display: block;
    font-size: calc(20px + 2vmin);
}

.beast-info {
    padding-left: 20px;
}

.abilities-list {
    list-style-type: none;
}