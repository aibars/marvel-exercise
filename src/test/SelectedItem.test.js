import React from 'react';
import { shallow } from 'enzyme';

import SelectedItem from '../components/SelectedItem';

describe('SelectedItem', () => {
    it('renders the card', () => {
        const component = shallow((
            <SelectedItem
                selectedItem={{
                    item: {
                        species: {
                            name: '1'
                        },
                        abilities: [
                            {
                                ability: "ability1"
                            }
                        ],
                        stats: [
                            { stat: "stat1" }
                        ],
                        sprites: {
                            front_default: 'sprite1'
                        }
                    },
                    isFetching: false
                }}
            />
        ));

        expect(component).toMatchSnapshot();
    });
});