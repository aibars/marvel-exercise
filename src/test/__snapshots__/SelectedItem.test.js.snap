// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItem renders the card 1`] = `ShallowWrapper {}`;
